### ISSUE TITLE
Random crash Cannot read property 'statusCode' of undefined

### ISSUE THREAD

**Issue by Freaders** [DoctorMcKay/node-steam-tradeoffer-manager#178]
So im getting this issue randomly. It is very rare. Happened twice out of 5k trades. What can cause that? Im using the latest version.
`
/bot/node_modules/steam-tradeoffer-manager/lib/classes/TradeOffer.js:478
                        if (response.statusCode == 401) {
                                    ^

TypeError: Cannot read property 'statusCode' of undefined
    at SteamCommunity.<anonymous> (/bot/node_modules/steam-tradeoffer-manager/lib/classes/TradeOffer.js:478:16)
    at Request._callback (/bot/node_modules/steamcommunity/components/http.js:67:15)
    at self.callback (/bot/node_modules/steamcommunity/node_modules/request/request.js:187:22)
    at emitOne (events.js:77:13)
    at Request.emit (events.js:169:7)
    at Request.onRequestError (/bot/node_modules/steamcommunity/node_modules/request/request.js:813:8)
    at emitOne (events.js:77:13)
    at ClientRequest.emit (events.js:169:7)
    at TLSSocket.socketErrorListener (_http_client.js:269:9)
    at emitOne (events.js:77:13)
    at TLSSocket.emit (events.js:169:7)
`

**Comment by Freaders** [comment]
so, got it now three times in a row.. i still dont know what causes it

**Comment by andrewda** [comment]
You should check if `response` is defined before checking the `statusCode`:

```javascript
if (response && response.statusCode == 401) {
    // your code
}

**Comment by Aareksio** [comment]
Update to the latest version. It's been fixed in 31e2b150e448034c2fb9251d356fe840aeba5009

**Comment by Freaders** [comment]
ohh, thanks a lot.. my mystake
